src/test/java/picard/illumina/parser/TileMetricsUtilTest.java

```diff
@@ -8,67 +8,90 @@
 import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Random;
 import java.util.stream.IntStream;
 
 public class TileMetricsUtilTest {
     @Test
     public void testFindTileMetrics() throws IOException {
-        Path baseDir = createTestDirs(0, true);
-        File tileMetricsFile = TileMetricsUtil.renderTileMetricsFileFromBasecallingDirectory(
+        List<Integer> populatedCycleDirs = new ArrayList<>();
+        Path baseDir = createTestDirs(0, populatedCycleDirs, true);
+        List<File> tileMetricsFiles = TileMetricsUtil.findTileMetricsFiles(
                 baseDir.toFile(), 0, false);
 
-        Assert.assertEquals(tileMetricsFile.getAbsolutePath(),
-                baseDir.resolve(TileMetricsUtil.INTEROP_SUBDIRECTORY_NAME)
-                        .resolve(TileMetricsUtil.TILE_METRICS_OUT_FILE_NAME).toString());
-
+        List<File> expected = generateFindTileMetricsExpected(baseDir, populatedCycleDirs, true);
+        Assert.assertEquals(tileMetricsFiles, expected);
         IOUtil.deleteDirectoryTree(baseDir.toFile());
     }
 
     @Test(expectedExceptions = IllegalStateException.class)
     public void testMissingTileMetrics() throws IOException {
-        Path baseDir = createTestDirs(0, false);
-        TileMetricsUtil.renderTileMetricsFileFromBasecallingDirectory(
+        List<Integer> populatedCycleDirs = new ArrayList<>();
+        Path baseDir = createTestDirs(0, populatedCycleDirs, false);
+        TileMetricsUtil.findTileMetricsFiles(
                 baseDir.toFile(), 0, false);
 
         IOUtil.deleteDirectoryTree(baseDir.toFile());
     }
 
-    // put tile metrics file in 1 random cycle dir and interop based ensure we get the one in base
+    // put tile metrics file in several permutations of random cycle dirs
+    // and make sure we get all metricsFiles each time
     @Test
-    public void testFindTileMetricsNovaSeqInBaseInterOp() throws IOException {
-        Path baseDir = createTestDirs(1, true);
-        File tileMetricsFile = TileMetricsUtil.renderTileMetricsFileFromBasecallingDirectory(
-                baseDir.toFile(), 1, true);
+    public void testFindTileMetricsNovaSeq() throws IOException {
+        int numCycles = 15;
+        int numRandomDirectoryPermutations = 5;
```

Comment:
Why is randomness necessary here? Or is this just instead of creating names by hand? In general randomness in a unit test is not great, since a test can pass and then fail based on a random number.